Tests/LibWeb/add_libweb_test.py

```diff
@@ -9,52 +9,120 @@
+    output_dir = output_prefix.parent
 
     # Create directories if they don't exist
     input_dir.mkdir(parents=True, exist_ok=True)
-    expected_dir.mkdir(parents=True, exist_ok=True)
+    output_dir.mkdir(parents=True, exist_ok=True)
 
     num_sub_levels = len(Path(test_name).parents) - 1
     path_to_include_js = "../" * num_sub_levels + "include.js"
 
-    # Create input and expected files
-    input_file.write_text(fR"""<!DOCTYPE html>
+    def generate_boilerplate():
+        """
+        return a tuple of strings with the input and
+        output boilerplate for a given test type
+        """
+
+        generic_boilerplate = R"""<!DOCTYPE html>
+<head>
+<style>
+</style>
+</head>
+<body>
+</body>
+"""
+
+        if test_type == "Text":
+            input_boilerplate = fR"""<!DOCTYPE html>
 <script src="{path_to_include_js}"></script>
 <script>
     {f"asyncTest(async (done)" if is_async else "test(()"} => {{
         println("Expected println() output");
     {f"    done();" if is_async else ""}
     }});
 </script>
-""")
-    expected_file.write_text("Expected println() output\n")
+"""
+            expected_boilerplate = "Expected println() output\n"
+
+        elif test_type == "Ref":
+            input_boilerplate = fR"""<!DOCTYPE html>
+<head>
+     <link rel="match" href="{"../" * num_sub_levels}../expected/{Path(test_name).with_suffix("")}-ref.html" />
+<style>
+</style>
+</head>
+<body>
+</body>
+"""
+
+            expected_boilerplate = f"Put equivalently rendering HTML for {test_name} here."
+
+        elif test_type == "Screenshot":
+            input_boilerplate = generic_boilerplate
+            expected_boilerplate = f"Put equivalently rendering HTML for {test_name} here."
+
+        # layout tests are async agnostic
+        elif test_type == "Layout":
+            input_boilerplate = generic_boilerplate
+            expected_boilerplate = f"""run
+ ./Meta/ladybird.sh run headless-browser --run-tests
+ "${{LADYBIRD_SOURCE_DIR}}/Tests/LibWeb" --rebaseline -f {input_file}
+to produce the expected output for this test
+"""
+            print("Delete <!DOCTYPE html> and replace it with <!--Quirks mode--> if test should run in quirks mode")
+
+        else:
+            # should be unreachable
+            raise ValueError(f"UNREACHABLE Invalid test type: {test_type}")
+
+        return (input_boilerplate, expected_boilerplate)
+
+    # Create input and expected files
+    input_boilerplate, expected_boilerplate = generate_boilerplate()
+    input_file.write_text(input_boilerplate)
+    output_file.write_text(expected_boilerplate)
 
     print(f"Text test '{Path(test_name).with_suffix('.html')}' created successfully.")
 
 
 def main():
+    supported_test_types = ["Ref", "Text", "Screenshot", "Layout"]
+
     parser = argparse.ArgumentParser(description="Create a new LibWeb Text test file.")
     parser.add_argument("test_name", type=str, help="Name of the test")
+    parser.add_argument("test_type", type=str,
```

Comment:
small nit, otherwise looks fine to my, but `argparse` has a `choices` field that can be used for the `supported_test_types`. Which would allow removing it from the description, and also the check of supported test types below

Reply:
Thanks!